packages/@uppy/react/src/DashboardModal.js

```diff
@@ -78,12 +148,85 @@ class DashboardModal extends React.Component {
 }
 
 DashboardModal.propTypes = {
+  uppy: uppyPropType.isRequired,
   target: typeof window !== 'undefined' ? PropTypes.instanceOf(window.HTMLElement) : PropTypes.any,
   open: PropTypes.bool,
   onRequestClose: PropTypes.func,
   closeModalOnClickOutside: PropTypes.bool,
   disablePageScrollWhenModalOpen: PropTypes.bool,
-  ...basePropTypes,
+  inline: PropTypes.bool,
+  plugins,
+  width: cssSize,
+  height: cssSize,
+  showProgressDetails: PropTypes.bool,
+  note: PropTypes.string,
+  metaFields,
+  proudlyDisplayPoweredByUppy: PropTypes.bool,
+  autoOpenFileEditor: PropTypes.bool,
+  animateOpenClose: PropTypes.bool,
+  browserBackButtonClose: PropTypes.bool,
+  closeAfterFinish: PropTypes.bool,
+  disableStatusBar: PropTypes.bool,
+  disableInformer: PropTypes.bool,
+  disableThumbnailGenerator: PropTypes.bool,
+  disableLocalFiles: PropTypes.bool,
+  disabled: PropTypes.bool,
+  hideCancelButton: PropTypes.bool,
+  hidePauseResumeButton: PropTypes.bool,
+  hideProgressAfterFinish: PropTypes.bool,
+  hideRetryButton: PropTypes.bool,
+  hideUploadButton: PropTypes.bool,
+  showLinkToFileUploadResult: PropTypes.bool,
+  showRemoveButtonAfterComplete: PropTypes.bool,
+  showSelectedFiles: PropTypes.bool,
+  waitForThumbnailsBeforeUpload: PropTypes.bool,
+  fileManagerSelectionType: PropTypes.string,
+  theme: PropTypes.string,
+  // pass-through to ThumbnailGenerator
+  thumbnailType: PropTypes.string,
+  thumbnailWidth: PropTypes.number,
+  locale,
+}
+// Must be kept in sync with @uppy/dashboard/src/Dashboard.jsx.
```

Comment:
Kudos for writing all these prop types. But yeah, this is why you want TS, this would be reused